presto-orc/src/main/java/com/facebook/presto/orc/AbstractOrcRecordReader.java

```diff
@@ -154,14 +155,23 @@ public AbstractOrcRecordReader(
 
         // reduce the included columns to the set that is also present
         ImmutableSet.Builder<Integer> presentColumns = ImmutableSet.builder();
-        ImmutableMap.Builder<Integer, Type> presentColumnsAndTypes = ImmutableMap.builder();
         OrcType root = types.get(0);
+        boolean onlyFixedWidthColumns = true;
+        int totalFixedWidth = 0;
         for (Map.Entry<Integer, Type> entry : includedColumns.entrySet()) {
             // an old file can have less columns since columns can be added
             // after the file was written
             if (entry.getKey() >= 0 && entry.getKey() < root.getFieldCount()) {
                 presentColumns.add(entry.getKey());
-                presentColumnsAndTypes.put(entry.getKey(), entry.getValue());
+
+                Type type = entry.getValue();
```

Comment:
Shall we move this logic out of this loop and move it closer to where it is used? Given it's in the constructor, having two for loops shouldn't be a performance concern. It will be clearer to separate the logic. Or, it could be even better to move the logic into another help like `adjustMaxBatchSizeBasedOnTypes` (something like this name).